<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

Exercise 1

In [ ]:
while True:
    print('*', end='')

# Interrupt the proceedings by pressing "Stop cell execution" on the left of the cell

Exercise 2

In [ ]:
print = 'PRINT'

In [ ]:
print('Hello World')

Exercise 3

In [ ]:
%%timeit
squares = [i**2 for i in range(10)]

2.22 μs ± 187 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)


In [ ]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

2.95 μs ± 120 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)


1. print(squares) gives [0, 1, 4, 9, 16, 25, 36, 49, 64, 81] for both cells
1. Execution time of
    1. First cell: 2.22 μs ± 187 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
    1. Second cell: 2.95 μs ± 120 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
1. What the numbers mean
    1.  Mean and standard deviation of execution time per loop, with mean in microseconds and std dev in nanoseconds.
    1.  7 runs is the default number of times the entire timing experiment was repeated.
    1. 100,000 loops are the default number of times the code inside the cell was executed per run. 
    

## Footnotes